src/librustc_trans/type_.rs

```diff
@@ -268,6 +269,20 @@ impl Type {
         }
     }
 
+    pub fn fat_ptr_data_type(&self) -> Type {
+        unsafe {
+            Type::from_ref(llvm::LLVMStructGetTypeAtIndex(self.to_ref(),
+                                                          abi::FAT_PTR_ADDR as u32))
+        }
+    }
+
+    pub fn fat_ptr_extra_type(&self) -> Type {
```

Comment:
There should already be a function to get this from `Ty` in `type_of` - if it's not public, it should be.